src/bindings/python/src/openvino/runtime/opset13/ops.py

```diff
@@ -222,3 +229,53 @@ def scaled_dot_product_attention(
         "causal": causal,
     }
     return _get_node_factory_opset13().create("ScaledDotProductAttention", inputs, attributes)
+
+
+@nameable_op
+def constant(
+    value: Union[NumericData, np.number, bool, np.bool_, list],
+    dtype: Union[NumericType, Type] = None,
+    name: Optional[str] = None,
+    *,
+    shared_memory: bool = False,
+) -> Constant:
+    """Create a Constant node from provided value.
+
+    :param value: One of: array of values or scalar to initialize node with.
+                  Note: if value is a scalar, shared memory is disabled.
+    :param dtype: The data type of provided data.
+                  If dtype does not match, data will be converted.
+                  Note: disables sharing of the memory when convertion occurs.
+    :param name: Optional name for output node.
+    :param shared_memory: keyword-only argument.
+                          If `True`, this Constant's memory is being shared with a host,
+                          that means the responsibility of keeping host memory is
+                          on the side of a user. Any action performed on the host
+                          memory is reflected on this Constant's memory!
+                          If `False`, data is being copied to this Constant.
+                          Requires data to be C_CONTIGUOUS if `True`.
+    :return: The Constant node initialized with provided data.
+    """
+    if isinstance(value, np.ndarray):
+        _value, _shared_memory = value, shared_memory
+    else:
+        _value, _shared_memory = np.array(value), False
```

Comment:
The new code is mostly related to handle this new `shared_memory` attribute. If `shared_memory==False`, usage of `make_constant_node` can be considered to ensure unified behaviour with current opset1.constant, while keeping the changes for the case if `shared_memory==True`.

https://github.com/openvinotoolkit/openvino/blob/369831caef55efe1f2d87889d3997f93c53ab6b8/src/bindings/python/src/openvino/runtime/utils/types.py#L140-L148

Reply:
I would disagree here, overall goal is to move to the new constructor as much as possible. It will put "into deprecation" older buffer-protocol approach and many overloaded constructors. At the end... most FEs are moving to this new constructor for Constant in many cases, for two reasons: first -- it's simpler, second -- shared memory feature.

I can agree that I need to look closer at https://github.com/openvinotoolkit/openvino/pull/21046#discussion_r1392924152 --  but at the same time I would redirect as many inputs as possible into new constructor.